### PR TITLE
Make AKO bootups faster

### PR DESCRIPTION
**Pull request by sudswasavi**
This commit introduces a change that works as follows:

- During bootup of AKO, we perform a full sync. This results
into the layer 2 models getting built before activating layer 3.
- The informer event handlers are then used to enqueue updates to
the layer 2 queues as well for the same updates which were processed
as a part of the full sync update.

This current fix optimizes the second point by caching the resource
version of the k8s objects during fullsync.

Then during the regular event updates, it compares the cached resource
versions in AKO with that of the event updated objects.

If the resource versions are same, then AKO does not enqueue the same
key again in layer 2, thus reducing the burden of layer 2 significantly
during bootup.

This fix potentially solves a starvation problem in AKO when during a
heavily loaded system, if AKO is rebooted, and some new routes/ingresses
are created, they are starved for these already pre-processed updates.

### CODE REVIEW

**Comment by monotosh-avi** [comment]
@chauhanshubham status update doesn't change the resource version, right ?

File: internal/k8s/ako_init.go [hunk 1]
```diff
@@ -269,9 +271,7 @@ func (c *AviController) InitController(informers K8sinformers, registeredInforme
 	}
 
 	// Setup and start event handlers for objects.
-	c.SetupEventHandlers(informers)
```
[the remark this answers is not in the document]
Comment (sudswasavi) [reply internal/k8s/ako_init.go:272]: Hope is all we have :-)

File: internal/k8s/ako_init.go [hunk 2]
```diff
@@ -585,6 +640,11 @@ func (c *AviController) FullSyncK8s() error {
 			}
 			for _, gwClassObj := range gwClassObjs {
 				key := lib.GatewayClass + "/" + utils.ObjKey(gwClassObj)
+				meta, err := meta.Accessor(gwClassObj)
+				if err == nil {
+					resVer := meta.GetResourceVersion()
+					objects.SharedResourceVerInstanceLister().Save(key, resVer)
+				}
```
[the remark this answers is not in the document]
Comment (sudswasavi) [reply internal/k8s/ako_init.go:647]: Good catch, done.

**Review by monotosh-avi — APPROVED** [review]

**Comment by sudswasavi** [comment]
build ako

**Review by chauhanshubham — APPROVED** [review]
lgtm. `AddCrdIndexer` can be an internal function. we can do that in a subsequent PR. thanks.

**Review by monotosh-avi — APPROVED** [review]

**Pull request merged**